aws/resource_aws_api_gateway_rest_api.go

```diff
@@ -410,7 +446,8 @@ func flattenApiGatewayEndpointConfiguration(endpointConfiguration *apigateway.En
 	}
 
 	m := map[string]interface{}{
-		"types": flattenStringList(endpointConfiguration.Types),
+		"types":            flattenStringList(endpointConfiguration.Types),
+		"vpc_endpoint_ids": flattenStringList(endpointConfiguration.VpcEndpointIds),
```

Comment:
This causes failures with other endpoints not of type `PRIVATE` that don't have VPC endpoint IDs. The failures look like this:
```
Error: error setting endpoint_configuration: Invalid address to set: []string{"endpoint_configuration", "0", "vpc_endpoint_ids"}

  on ../modules/api_gateway_domain/main.tf line 5, in resource "aws_api_gateway_domain_name" "api_gateway_domain":
   5: resource "aws_api_gateway_domain_name" "api_gateway_domain" {
```

Tested alternative implementation:
```go
	m := map[string]interface{}{
		"types": flattenStringList(endpointConfiguration.Types),
	}

	if len(endpointConfiguration.VpcEndpointIds) > 0 {
		m["vpc_endpoint_ids"] = flattenStringList(endpointConfiguration.VpcEndpointIds)
	}
```

Reply:
i missed that this function also affects the domain resource

Reply:
fixed!